src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -548,7 +548,7 @@ class Direct$Type$Buffer$RW$$BO$
         assert (pos <= lim);
         int rem = (pos <= lim ? lim - pos : 0);
         try {
-            SCOPED_MEMORY_ACCESS.copyMemory(scope(), null, null,
+            SCOPED_MEMORY_ACCESS.copyMemory(scope(), scope(), null,
```

Comment:
Is this deliberate? same strange to use same scope for both address and target?

Reply:
Yes, it is. In this case the buffer is copying to itself, moving the data down to the beginning of the buffer, position zero.